paperqa/sources/clinical_trials.py

```diff
@@ -0,0 +1,321 @@
+import json
+import logging
+from contextlib import suppress
+from datetime import datetime
+from typing import Any
+
+import aiohttp
+from aiohttp import ClientResponseError, ClientSession
+from aiohttp.web import HTTPBadRequest
+from tenacity import (
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_incrementing,
+)
+
+from paperqa.docs import Docs
+from paperqa.settings import Settings
+from paperqa.types import DocDetails, Embeddable, Text
+from paperqa.utils import gather_with_concurrency
+
+logger = logging.getLogger(__name__)
+
+
+CLINICAL_TRIALS_BASE = "clinicaltrials.gov"
+CLINICAL_TRIALS_URL = f"https://{CLINICAL_TRIALS_BASE}"
+STUDIES_API_URL = CLINICAL_TRIALS_URL + "/api/v2/studies"
+SEARCH_API_FIELDS = "NCTId,OfficialTitle"
+SEARCH_PAGE_SIZE = 1000
+TRIAL_API_FIELDS = "protocolSection,derivedSection"
+DOWNLOAD_CONCURRENCY = 20
+TRIAL_CHAR_TRUNCATION_SIZE = 30_000  # larger will prevent embeddings from working
+MALFORMATTED_QUERY_STATUS: int = 400
+
+
+@retry(
+    stop=stop_after_attempt(3),
+    wait=wait_incrementing(0.1, 0.1),
+    retry=retry_if_exception_type(ClientResponseError),
+)
+async def api_search_clinical_trials(query: str, session: ClientSession) -> dict:
+    async with session.get(
+        STUDIES_API_URL,
+        params={
+            "query.term": query,
+            "fields": SEARCH_API_FIELDS,
+            "pageSize": SEARCH_PAGE_SIZE,
+            "countTotal": "true",
+            "sort": "@relevance",
+        },
+    ) as response:
+        if response.status == MALFORMATTED_QUERY_STATUS:
+            # the 400s from clinicaltrials.gov are not JSON
```

Comment:
Do you mind moving this comment to live with `MALFORMATTED_QUERY_STATUS`, or renaming `MALFORMATTED_QUERY_STATUS` to be `CLINICAL_TRIALS_DIDNT_GIVE_JSON = 400`